client/app/components/packages/applicant-fieldset.js

```diff
@@ -1,7 +1,32 @@
 import Component from '@glimmer/component';
 import { action } from '@ember/object';
+import { STATE_OPTION_SET } from '../../models/applicant';
 
 export default class ApplicantFieldset extends Component {
+  get stateOptions() {
+    return Object.values(STATE_OPTION_SET);
+  }
+
+  // TODO: Use a global helper instead after writing
+  // unified optionset handling
+  get stateLabelLookup() {
+    const lookup = {};
+    const { stateOptions } = this;
+
+    for (let i = 0; i < stateOptions.length; i += 1) {
+      lookup[stateOptions[i].code] = stateOptions[i];
+    }
+
+    return lookup;
+  }
+
+  // TODO: Refactor this to engage with Changeset, when changesets are
+  // introduced for applicant, BBls.
+  @action
+  setApplicantState({ code: stateCode }) {
+    this.args.applicant.dcpState = stateCode;
+  }
+
```

Comment:
there's already an `updateAttr` method below. Why make a new one?

Reply:
Is there a way to destructure the selected object before passing it into `newVal` parameter of `updateAttr`?

Reply:
I'm not sure — have you looked at the docs? Could also just call `updateAttr` for here.

Not sure `setApplicantState` is the best name. "State" means a lot of different things in different contexts. If you need to pull an attribute from the selected state object, maybe that's what this method could be responsible for?

Reply:
Made some changes!